### PR TITLE
RELATED: ONE-3393 Integration of color mapping to AD

### PR DESCRIPTION
**Pull request by pbenes**

### CODE REVIEW

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/2111/ : SUCCESS in 19s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/1564/ : FAILURE in 15m 54s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/2116/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/1569/ : FAILURE in 3m 28s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/2118/ : SUCCESS in 39s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/1571/ : SUCCESS in 6m 11s

File: src/components/visualizations/chart/colorFactory.ts [hunk 1]
```diff
@@ -99,14 +108,14 @@ export abstract class ColorStrategy implements IColorStrategy {
...
-    protected abstract createColorMapping(
+    protected abstract createColorAssignment(
         colorPalette: IColorPalette,
         colorMapping: IColorMapping[],
         measureGroup: MeasureGroupType,
         viewByAttribute: any,
         stackByAttribute: any,
         afm: AFM.IAfm
-    ): IColorAssignment[];
+    ): { fullColorAssignment: IColorAssignment[], outputColorAssignment?: IColorAssignment[] };
```
Comment (ivanmjartan) [inline src/components/visualizations/chart/colorFactory.ts]: Define interface for return type { fullColorAssignment: IColorAssignment[], outputColorAssignment?: IColorAssignment[] } something like  IColorAssignmentType or wrapper or ...

File: src/components/visualizations/chart/colorFactory.ts [hunk 2]
```diff
@@ -66,24 +69,30 @@ export abstract class ColorStrategy implements IColorStrategy {
...
             afm
         );
+        this.fullColorAssignment = fullColorAssignment;
+        this.outputColorAssignment = outputColorAssignment ? outputColorAssignment : fullColorAssignment;
 
-        this.palette = this.createPalette(colorPalette, this.colorAssignment, viewByAttribute, stackByAttribute);
+        this.palette = this.createPalette(colorPalette, this.fullColorAssignment, viewByAttribute, stackByAttribute);
     }
 
     public getColorByIndex(index: number): string {
```
Comment (ivanmjartan) [inline src/components/visualizations/chart/colorFactory.ts]: Move function getColorFromMapping(mappingHeader: IMappingHeader, colorMapping: IColorMapping[]
    ): IColorItem 

export function getColorByGuid(colorPalette: IColorPalette, guid: string, index: number) 

function getRgbStringFromRGB(color: IRGBColor) {

to colors utils

File: src/components/visualizations/chart/colorFactory.ts [hunk 3]
```diff
@@ -132,27 +141,36 @@ function getRgbStringFromRGB(color: IRGBColor) {
...
+        };
     }
 
     private mapColorsFromMeasures(
         measureGroup: MeasureGroupType,
         afm: AFM.IAfm,
         colorMapping: IColorMapping[],
         colorPalette: IColorPalette
-    ): IColorAssignment[] {
+    ): { allMeasuresAssignment: IColorAssignment[], pureMeasuresAssignment: IColorAssignment[] } {
```
Comment (ivanmjartan) [inline src/components/visualizations/chart/colorFactory.ts]: rename pureMeasuresAssignment to nonDerivedAssignment

File: src/components/visualizations/chart/colorFactory.ts [hunk 4]
```diff
@@ -234,10 +256,13 @@ function getAttributeColorMapping(
         colorMapping: IColorMapping[]
     ): IColorAssignment[] {
     let currentColorPaletteIndex = 0;
-    return attribute.items.map((headerItem: any) => {
+    const uniqItems: Execution.IResultAttributeHeaderItem[]
```
Comment (ivanmjartan) [inline src/components/visualizations/chart/colorFactory.ts]: `if (parentMeasureIndex > -1) {
                 const sourceMeasureColor = measuresColorAssignment[parentMeasureIndex].color;
                 const rgbColor = isGuidColorItem(sourceMeasureColor)
                     ? getColorByGuid(colorPalette, sourceMeasureColor.value, measureItemIndex)
                     : sourceMeasureColor.value;
                 return {
                     ...mapItem,
                     color: {
                         type: 'rgb' as RGBType,
                         value: getLighterColorFromRGB(rgbColor, 0.6)
                     }
                 };
             }
             return {
                 ...mapItem,
                 color: mapItem.color
             };`

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/2125/ : SUCCESS in 38s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/1576/ : SUCCESS in 5m 36s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/2126/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/1577/ : SUCCESS in 5m 57s

**Comment by peterlacko** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/2129/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/1579/ : SUCCESS in 7m 06s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/859/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/2130/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/484/ : UNSTABLE in 14m 15s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/2131/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/1580/ : SUCCESS in 1m 59s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/2136/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/1585/ : SUCCESS in 9m 42s

**Comment by peterlacko** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/2142/ : SUCCESS in 9s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/1592/ : SUCCESS in 7m 46s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/867/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/2143/ : SUCCESS in 9s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/488/ : SUCCESS in 20m 08s

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-merge-compliance-check/487/ : SUCCESS in 18s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/1607/ : SUCCESS in 3m 59s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/873/ : SUCCESS in 15m 14s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-examples-zuul-docker/431/ : SUCCESS in 17m 43s

**Pull request merged**